ports/sophus/portfile.cmake

```
@@ -1,16 +1,15 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO strasdat/Sophus
-    REF 1.22.10
-    SHA512 0b212a015d487b3a39a9c4beb4280ebd659782f527ec863f429b5aa57462d53c41976c9285894855b380baa97d5fda35b46dfd09aa31653fb35b51039816ac78
-    HEAD_REF master
+    REF 1.24.6-rc1
+    SHA512 c1ba40b823cabce3fe83f528837ac111f4d746d6679fb920abd7be32de149c0937bb9a5049da156aec28a9b9fedbebae76a056de12707c01c5cb40dc9197c3e4
+    HEAD_REF main
 )
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS 
-        -DBUILD_SOPHUS_TESTS=OFF
-        -DBUILD_SOPHUS_EXAMPLES=OFF
+        -DBUILD_SOPHUS_TESTS=ON
```

```suggestion
        -DBUILD_SOPHUS_TESTS=OFF
```
https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#do-not-build-testsdocsexamples-by-default

---

@dg0yt sophus is a header only library and if we do not build the tests we actually do not know whether Sophus works in the target OS.

---

It is not the first header-only lib in vcpkg... Which properties do really need to be tested *in every build* and isn't sufficiently controlled by upstream tests before a release?

Note that vcpkg provides build recipes and binary caching, but it doesn't distribute binary artifacts. Ports are built more often then one would expect.

---

Hm, yeah, that is an option. All windows builds were failing and I could only detect that with in vcpkg CI, since there are currently no windows builds upstream. But yeah we do not need to have that verified in every build.

---

What may help until there is another downstream port is to add tiny test port in `scripts/test-ports`.
There is already a small collection. Some do only pull in non-default feautures, but other really build test projects (e.g. vcpkg-ci-soci).
This gives an end-to-end test experience in vcpkg CI (or for manual tests) without affecting every build.